### ISSUE TITLE
Gitlab support

### ISSUE THREAD

**Issue by gsemet** [google/copybara#153]
Hi. 

I wonder if a Gitlab integration would be imaginable somewhere in the future, especially to mirror incoming public contribution to merge request on the internal SoT, and closing them elegantly? Look like the integration on Gerrit and Github and pretty fine, but on other systems we need to do some manual cleanup.

Thanks

**Comment by mikelalcon** [comment]
Hi,

Yes, Copybara architecture is thought of so that new types of
components can be added (Like Gitlab). We would be happy to accept
contributions

Cheers,
Mikel.

**Comment by sluongng** [comment]
@mikelalcon what is your opinion regarding using something like https://github.com/gitlab4j/gitlab4j-api (MIT license) instead of having to implement the API client from scratch like what you have here for Github? That could save the implementation effort quite a bit.

**Comment by hsudhof** [comment]
No, regrettably we had no cycles to work on this. As Mikel mentioned, this would look like the GitHub setup, with the same extension points for alternate transports etc if anybody want to send a PR.
